Gems/Streamer/StreamerProfiler/Code/Source/StreamerProfilerSystemComponent.cpp

```diff
@@ -0,0 +1,776 @@
+    }
+
+    void StreamerProfilerSystemComponent::DrawFileLocks(AZ::IO::IStreamer& streamer)
+    {
+#if defined(IMGUI_ENABLED)
+        // Queue next request for update if needed.
+        if (m_readingFileLocks)
+        {
+            AZ::IO::FileRequestPtr request = streamer.Report(*m_readingFileLocks, AZ::IO::IStreamerTypes::ReportType::FileLocks);
+            auto callback = [this, readingFileLocks = m_readingFileLocks](AZ::IO::FileRequestHandle)
+            {
+                // Once the request has been completed, enable the stats to draw.
+                StatsContainer* displayingLocks = nullptr;
+                while (!m_transferFileLocks.compare_exchange_strong(displayingLocks, readingFileLocks))
+                {
+                    displayingLocks = nullptr;
+                }
+            };
+            streamer.SetRequestCompleteCallback(request, AZStd::move(callback));
+            streamer.QueueRequest(request);
+
+            m_readingFileLocks = nullptr;      
+        }
+
+        // Try to get a handle to the fresh stats.
+        StatsContainer* transferFileLocks = m_transferFileLocks;
+        while (!m_transferFileLocks.compare_exchange_strong(transferFileLocks, nullptr))
+        {
+            transferFileLocks = m_transferFileLocks;
+        }
+        if (transferFileLocks)
+        {
+            // Rotate the buffers
+            m_displayingFileLocks->clear();
+            m_readingFileLocks = m_displayingFileLocks;
+            m_displayingFileLocks = transferFileLocks;
+        }
+
+        // Draw the display list. One should always be assigned.
+        AZ_Assert(m_displayingFileLocks, "Expected to always have a valid list of file locks, even if it's empty.");
+        ImGui::Text("Total file lock count: %llu", m_displayingFileLocks->size());
+        if (ImGui::Button("Flush all"))
+        {
+            streamer.QueueRequest(streamer.FlushCaches());
+        }
+        if (ImGui::BeginTable("File Locks", 3, ImGuiTableFlags_Borders | ImGuiTableFlags_Resizable))
+        {
+            ImGui::TableSetupColumn("Node", ImGuiTableColumnFlags_WidthStretch, 0.20f);
+            ImGui::TableSetupColumn("File", ImGuiTableColumnFlags_WidthStretch, 0.70f);
+            ImGui::TableSetupColumn("Flush", ImGuiTableColumnFlags_WidthStretch, 0.10f);
+            ImGui::TableHeadersRow();
+
+            for (AZ::IO::Statistic& stat : *m_displayingFileLocks)
+            {
+                ImGui::TableNextRow();
+
+                ImGui::TableNextColumn();
+                ImGui::Text("%.*s", AZ_STRING_ARG(stat.GetOwner()));
+
+                ImGui::TableNextColumn();
+                DrawStatisticValue(stat.GetValue());
+
+                const AZStd::string* path = AZStd::get_if<AZStd::string>(&stat.GetValue());
+                if (path)
+                {
+                    ImGui::TableNextColumn();
+                    AZStd::fixed_string<1024> name = "Flush##";
+                    name += *path;
```

Comment:
Is it possible for the path to exceed 1024-len('Flush##') ? Maybe truncate to a fixed length as well to prevent possible overflow

Reply:
`fixed_string` would handle that internally plus the path has a limit as well, though that's a bit lost by the variables being passed around. I added a bit of extra code to truncate the path if it exceeds the total limit.

Reply:
Well realistically, this will not be a problem on Widnows since paths would have to be below 260, even if the file system starts supporting paths above that, most applications wouldn't.